src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -213,6 +213,17 @@ impl TargetInfo {
                 should_replace_hyphens: false,
             })
         }
+        else if target_triple.ends_with("windows-gnu")
+                && crate_type.ends_with("dylib")
+                && suffix == ".dll"
+        {
+            ret.push(FileType {
+                suffix: ".dll.a".to_string(),
+                prefix: "lib".to_string(),
```

Comment:
```suggestion
                prefix: prefix.clone(),
```
There is no need to hardcode `lib` prefix because it works in either form.